### DISCUSSION TITLE
[V5] Get data from a relationship inside a pivot table

### DISCUSSION THREAD

**Discussion by mathDias** [adonisjs/core#3219] [Help]
Hi everyone.
Is it possible to get the data related with a pivot table in a Many to Many relationship? 

For example:
I have a car model related by many to many relationship with categories.
Some of these cars of some of these categories, (so the union of the both), have specifcs characteristics, today I'm using other table related with the pivot table of the first many to many relationship. Is it possible to get the data from the relation with the pivot table? Today I can get only values by the pivotColumns configuration, but I need to get the relationship and not only static values.

**Comment by GabrielNBDS** [comment]
Set the [pivot columns in your model](https://docs.adonisjs.com/reference/orm/relations/many-to-many#pivotcolumns)

And then this can be made:
```js
.preload('yourRelationship', (query) => {
    query.pivotColumns(['yourPivotColumn'])
})
```

The pivot columns values will be in `$extras`

**Reply by mathDias** [reply]
Yeah, it's ok. But if i need to get a data from a relationship with the pivot table, is it possible to do?
For example:
`.preload('manyToManyRelationship', (query) => {
    query.preload('relationshipOfpivot table')...
})`

**Reply by GabrielNBDS** [reply]
I didnt get it.

lets say you have this many to many relation:
```
+-------------+--------------+----------------------------+
|                        category                         |
+-------------+--------------+----------------------------+
| id          | int(11)      | PRIMARY KEY AUTO_INCREMENT |
| name        | varchar(255) |                            |
+-------------+--------------+----------------------------+

+-------------+--------------+----------------------------+
|                        question                         |
+-------------+--------------+----------------------------+
| id          | int(11)      | PRIMARY KEY AUTO_INCREMENT |
| title       | varchar(255) |                            |
| text        | varchar(255) |                            |
+-------------+--------------+----------------------------+

+-------------+--------------+----------------------------+
|              question_categories_category               |
+-------------+--------------+----------------------------+
| questionId  | int(11)      | PRIMARY KEY FOREIGN KEY    |
| categoryId  | int(11)      | PRIMARY KEY FOREIGN KEY    |
| anything    | varchar(255) |                            |
+-------------+--------------+----------------------------+
```

Do you want to get the `anything` column from the pivot table or you want to get categories from question (or vice versa) or what you want is another thing?

**Reply by Ralireza** [reply]
how to get anything??

**Reply by sdivyanshu112** [reply]
> I didnt get it.
> 
> lets say you have this many to many relation:
> 
> ```
> +-------------+--------------+----------------------------+
> |                        category                         |
> +-------------+--------------+----------------------------+
> | id          | int(11)      | PRIMARY KEY AUTO_INCREMENT |
> | name        | varchar(255) |                            |
> +-------------+--------------+----------------------------+
> 
> +-------------+--------------+----------------------------+
> |                        question                         |
> +-------------+--------------+----------------------------+
> | id          | int(11)      | PRIMARY KEY AUTO_INCREMENT |
> | title       | varchar(255) |                            |
> | text        | varchar(255) |                            |
> +-------------+--------------+----------------------------+
> 
> +-------------+--------------+----------------------------+
> |              question_categories_category               |
> +-------------+--------------+----------------------------+
> | questionId  | int(11)      | PRIMARY KEY FOREIGN KEY    |
> | categoryId  | int(11)      | PRIMARY KEY FOREIGN KEY    |
> | anything    | varchar(255) |                            |
> +-------------+--------------+----------------------------+
> ```
> 
> Do you want to get the `anything` column from the pivot table or you want to get categories from question (or vice versa) or what you want is another thing?

want anything data

**Reply by NishadVadgama** [reply]
I had been on the same situation where I needed to access `anything` column. The way I got to is is using `$extras.pivot_anything`. any columns used in `pivotColumns` will prefix `pivot_` for a column
